keps/sig-scheduling/5004-dra-extended-resource/README.md

```diff
@@ -0,0 +1,989 @@
+tests](https://github.com/kubernetes/kubernetes/commit/cecebe8ea2feee856bc7a62f4c16711ee8a5f5d9)
+as part of the normal Kubernetes "integration" jobs which cover [the dynamic
+resource
+controller](https://github.com/kubernetes/kubernetes/blob/294bde0079a0d56099cf8b8cf558e3ae7230de12/test/integration/scheduler_perf/util.go#L135-L139).
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+End-to-end testing depends on a working resource driver and a container runtime
+with CDI support. A [test
+driver](https://github.com/kubernetes/kubernetes/tree/master/test/e2e/dra/test-driver)
+was developed as part of the overall DRA development effort. We will add tests to
+ensure `ExtendedResourceName`s are handled by the scheduler as described in this KEP.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback
+- Additional tests are in Testgrid and linked in KEP
+
+#### GA
+
+- 3 examples of vendors making use of the extensions proposed in this KEP
+- Scalability tests that mirror real-world usage as determined by user feedback
+- Allowing time for feedback
+
+### Upgrade / Downgrade Strategy
+
+The usual Kubernetes upgrade and downgrade strategy applies for in-tree
+components. Vendors must take care that upgrades and downgrades work with the
+drivers that they provide to customers.
+
+### Version Skew Strategy
+
+All of the API extensions proposed in this KEP is the optional
+`ExtendedResourceName` in `DeviceClass`, and `ExtendedResourceClaimStatus` in
+`Pod`. There is no risk for version skew downgrades
+because these `DeviceClass` and `Pod` will never have existed in
+older clusters.
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: DRAExtendedResource
+  - Components depending on the feature gate:
+    - kube-apiserver
+    - kube-scheduler
+    - kubelet
+
+###### Does enabling the feature change any default behavior?
+
+No
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+
+Yes. Applications that were already deployed and are running will continue to
+work. They will continue to work when restarted because the CDI devices that
+have been prepared for them won't change across the restart.
+
+The DRA driver itself should also be able to survive a rollback, as there is no
+DRA driver change in this KEP.
```

Comment:
When the pod is deleted, and the node unprepare resources is called, will it work as expected?

Reply:
Yes, there is no difference in this respect between a reguar claim, and the special claim backed by DRA. They are handled the same way in kubelet.